tests/unit/projection-test.js

```diff
@@ -0,0 +1,1464 @@
+const NORM_PROJECTED_AUTHOR_CLASS = 'com.example.bookstore.projected-type.projected-author';
+const PUBLISHER_CLASS = 'com.example.bookstore.publisher';
+const PROJECTED_PUBLISHER_CLASS = 'com.example.bookstore.projectedType.projectedPublisher';
+const NORM_PROJECTED_PUBLISHER_CLASS = 'com.example.bookstore.projected-type.projected-publisher';
+
+module('unit/projection', function(hooks) {
+  setupTest(hooks);
+
+  hooks.beforeEach(function() {
+    initializeStore(this);
+
+    this.store = function() {
+      return this.owner.lookup('service:store');
+    };
+
+    SchemaManager.registerSchema({
+      modelIsProjection(modelName) {
+        return /^com\.example\.bookstore\.projection\./i.test(modelName);
+      },
+
+      includesModel(modelName) {
+        return /^com\.example\.bookstore\./i.test(modelName);
+      },
+
+      computeBaseModelName(projectionModelName) {
+        let modelSchema = this.models[projectionModelName];
+        return modelSchema && modelSchema.projectedType;
+      },
+
+      computeAttributeReference(key, value, modelName) {
+        if (/^isbn:/.test(value)) {
+          return {
+            id: value,
+            type: BOOK_CLASS_PATH,
+          }
+        } else if (/^urn:([^:]+):(.*)/.test(value)) {
+          let parts = /^urn:([^:]+):(.*)/.exec(value);
+          let type = parts[1];
+          let modelSchema = this.models[modelName];
+          if (modelSchema && modelSchema.resolvedTypes && modelSchema.resolvedTypes[key]) {
+            type = modelSchema.resolvedTypes[key];
+          }
+          return {
+            type,
+            id: parts[2],
+          };
+        }
+      },
+
+      isAttributeArrayReference(key) {
+        return key === 'otherBooksInSeries';
+      },
+
+      computeNestedModel(key, value, modelName) {
+        if (!value || typeof value !== 'object' || value.constructor === Date) {
+          return null;
+        }
+        let valueType = value.type;
+        let modelSchema = this.models[modelName];
+        if (modelSchema && modelSchema.resolvedTypes && modelSchema.resolvedTypes[key]) {
+          valueType = modelSchema.resolvedTypes[key];
+        }
+        return {
+          type: valueType,
+          id: value.id,
+          attributes: value,
+        }
+      },
+
+      models: {
+        [NORM_BOOK_CLASS_PATH]: {
+          aliases: {
+            name: 'title',
+            cost: 'price',
+            pub: 'publisher',
+            releaseDate: 'pubDate',
+            pb: 'paperback',
+            hb: 'hardback',
+          },
+          defaults: {
+            publisher: 'Penguin Classics',
+            hardback: true,
+            paperback: true,
+            publishedIn: 'US',
+          },
+          transforms: {
+            // This interferes with the URN resolution
+            // publisher(value) {
+            //   return `${value}, of course`;
+            // },
```

Comment:
why is this dead code here?

is there a problem that needs to be resolved?

Reply:
@runspired I've commented this out to get a test passing, but I think I've left the code there if you have are thinking of a case, which I was missing.

Reply:
it was copy-pasta from David's other tests that should be culled